### DISCUSSION TITLE
plotting RGB image of the entire Sentinel2 TILE

### DISCUSSION THREAD

**Discussion by roybenhayun** [torchgeo/torchgeo#1599] [Q&A] [answered]
the Sentinel2 dataset can render the RGB image of a single **sample**. at that point the sample already contains pytorch tensors of a piece of the entire Sentinel2 TILE.

we would like to render the RGB image of the **entire TILE**, once before the actual training starts. 
guessing that it should load RGB bands from the .jp2 files (under \GRANULE\...\IMG_DATA\), then some conversions may occur, then put them into tensors, then some tweaks for matplotlib etc. there could be some work involved.

is there a way in TorchGeo for that? 
or possibly can someone share code that does this? 

thanks in advance!

**Comment by adamjstewart** [comment, marked as answer]
Our `dataset.plot` methods can only accept a sample as input. You _could_ set the size to be the size of an entire tile if you want. You'll likely need to downsample the resolution, otherwise the image is too large to plot.

**Comment by roybenhayun** [comment]
ok. see working code using the Dataset and [PreChippedGeoSampler](https://torchgeo.readthedocs.io/en/stable/api/samplers.html#pre-chipped-geo-sampler).
other then being slow, look reasonable? any suggestion?

```
    ds = Sentinel2(r'SENTINEL2.SAFE')
    print(f"ds: {ds}")  
    width = ds.bounds.maxx - ds.bounds.minx
    height = ds.bounds.maxy - ds.bounds.miny
    print(f"width: {width}, height: {height}")

    fig, axs = plt.subplots(nrows=1, ncols=2, figsize=(2 * 5, 5))

    axs[0].set_title("full input image")
    full_ds_sampler = PreChippedGeoSampler(ds)
    dataloader = DataLoader(ds, sampler=full_ds_sampler, collate_fn=stack_samples)
    print(f"dataloader: {dataloader}")
    for sample in dataloader:
        image = sample["image"]/10000.0
        print(f"image: {image.shape}")
    permuted_image = image.cpu()[0][1:4, :, :].flip(0).permute(1, 2, 0)
    print(f"permuted_image: {permuted_image.shape}")
    axs[0].imshow(permuted_image)

    axs[1].set_title("cropped input image")
    bbox = BoundingBox(ds.bounds.minx + width / 4, ds.bounds.minx + (3 * width / 4),
                            ds.bounds.miny + height / 3, ds.bounds.miny + (2 * height / 3),
                            ds.bounds.mint, ds.bounds.maxt)
    chip_sampler = PreChippedGeoSampler(ds, bbox)
    dataloader = DataLoader(ds, sampler=chip_sampler, collate_fn=stack_samples)
    print(f"dataloader: {dataloader}")
    for sample in dataloader:
        image = sample["image"]/10000.0
        print(f"chip image: {image.shape}")
    permuted_chip_image = image.cpu()[0][1:4, :, :].flip(0).permute(1, 2, 0)
    axs[1].imshow(permuted_chip_image)
    plt.tight_layout()
    plt.show()
    plt.close()

```

**Reply by adamjstewart** [reply]
It's slow because each image is massive, if you use:
```python
ds = Sentinel2('SENTINEL2.SAFE', res=1000)
```
it will be much faster. Then you don't have to bother with `bbox` cropping.

**Reply by roybenhayun** [reply]
the bbox cropping is for other purposes that do require cropping. it sneaked into the code...

the res=1000 is fantastic! saves lots of time. many thanks for this tip.
